src/Request.ts

```diff
@@ -0,0 +1,117 @@
+import https from 'https'
+import http from 'http'
+import concat from 'concat-stream'
+import url from 'url'
+import FormData from 'form-data'
+import { UploadError, UploadErrorCode } from './UploadError'
+
+type SourceMapPayload = BrowserSourceMapPayload
+
+export interface BrowserSourceMapPayload {
+  apiKey: string
+  appVersion?: string
+  minifiedUrl: string
+  sourceMap: { filepath: string, data: string }
+  minifiedFile?: { filepath: string, data: string }
+  overwrite?: boolean
+}
+
+const MAX_ATTEMPTS = 5
+const RETRY_INTERVAL = parseInt(process.env.BUGSNAG_RETRY_INTERVAL as string) || 1000
+const TIMEOUT = parseInt(process.env.BUGSNAG_TIMEOUT as string) || 30000
+
+export default async function request (endpoint: string, payload: SourceMapPayload, requestOpts: http.RequestOptions): Promise<void> {
+  let attempts = 0
+  const go = async (): Promise<void> => {
+    try {
+      attempts++
+      await send(endpoint, payload, requestOpts)
+    } catch (err) {
+      if (err && err.isRetryable !== false && attempts < MAX_ATTEMPTS) {
+        await new Promise((resolve) => setTimeout(resolve, RETRY_INTERVAL))
+        return await go()
+      }
+      throw err
+    }
+  }
+  await go()
+}
+
+function createFormData (payload: SourceMapPayload): FormData {
```

Comment:
It looks like this is the only place that actually uses `payload`, so maybe we can pass this function as a callback, to let each module add its own stuff — can probably wait until we need it though